### ISSUE TITLE
Json Schema draft -04 and JSR 303 annotations

### ISSUE THREAD

**Issue by joshi-hitesh** [mulesoft-labs/raml-for-jax-rs#157]
I am using  $schema": "http://json-schema.org/draft-04/schema#" And use my plugin to generate POJO with JSr 303 annotations enabled.
    useJsr303Annotations true

however, the generated POJO does not have JSR annotations for the required fields.

Fields that are mandatory in JSON schema do not have @NotNull in code.
This works perfectly with JSON schema  draft -03. Please help.

**Comment by joshi-hitesh** [comment]
I tried generating the schema directly via jsonschema2pojo.org with Json schema version 4.It works well and generates JSR303 annotations.
It doesn't work here.Is it because this plugin excludes following dependency in pom
 <exclusions>
            	< exclusion >
            		< artifactId >jackson-databind< /artifactId >
            		< groupId >com.fasterxml.jackson.core</ groupId >
            	</ exclusion >
            </ exclusions >

**Comment by joshi-hitesh** [comment]
My bad, jsonschema2pojo supports generating JSR 303 annotations,and that is exactly what this plugin uses- Context.java
 final GenerationConfig jsonSchemaGenerationConfig = configuration.createJsonSchemaGenerationConfig();
        schemaMapper = new SchemaMapper(new RuleFactory(jsonSchemaGenerationConfig, getAnnotator(jsonSchemaGenerationConfig),
                new SchemaStore()), new SchemaGenerator());

I have tried generating the Model objects with jsonSchema2Pojo directly and it works great.

However, RAML plugin uses an older version of the plugin 0.4.11. And also excludes the following dependecy com.fasterxml.jackson.core.

I tried generating the Model with older version too, it only works if I remove the exclusion. Still not sure why @NotNull is not getting generated.
